### DISCUSSION TITLE
Simple exemple to start

### DISCUSSION THREAD

**Discussion by lugnier** [seraphx2/ESI.NET#63] [Troubleshooting]
Hi, 
I'm a wpf developer since 25 years.
Could you provide a simple exemple to connect, ask the sso logon and retrieve something simple like caracter information.
I already have an application declaration (secretid,...)

Thank you very much

Benoist

**Comment by lugnier** [comment]
I tried this code but the .response never end
In my code, the response is "False"

Thanks

IOptions<EsiConfig> config;

        public MainWindow()
        {
            InitializeComponent();

            config = Options.Create(new EsiConfig()
            {
                EsiUrl = "https://esi.evetech.net/",
                DataSource = DataSource.Tranquility,
                ClientId = "xxx",
                SecretKey = "yyy",
                CallbackUrl = "",
                UserAgent = "Obilan"
            });
        }

      
        private void Button_Click(object sender, RoutedEventArgs e)
        {
            var r = new EsiClient(config);
            var response = r.Universe.Systems().Wait(10000);
            Debugger.Break();
        }

**Comment by lugnier** [comment]
my update
This is working for public informations
      static async Task Main(string[] args)
        {

            //IOptions config;
            var config = Options.Create(new EsiConfig()
            {
                EsiUrl = "https://esi.evetech.net/",
                DataSource = DataSource.Tranquility,
                ClientId = "598d271e4a844d6caf975fca66ee4c0d",
                SecretKey = "y255ai5PDOLYGLvGIdbsA7didd7ItuM3EiZzdaj9",
                CallbackUrl = "http://localhost/oauth-callback",
                UserAgent = "Obilan"
            });

            EsiClient _client = new EsiClient(config);  
            //var response = await _client.Universe.Systems();
            //Debugger.Break();

            var response = _client.Universe.Names(new List<long>()
{
    1590304510,
    99006319,
    20000006
}).Result;

but I can't get SSO login
I don't understand what is the code in 

SsoToken token = await _client.SSO.GetToken(GrantType.AuthorizationCode, code);

**Comment by seraphx2** [comment]
OK, cool. You figured out the async part.
Are you familiar with OAuth at all?
The Readme links to a page though should help with understanding that flow.

**Reply by lugnier** [reply]
Thanks for your answer. I'm not familiar with oauth and all my reading hav'nt help for this part. I'll try to read it more carefully. I hoped that you had a simple code to get something simple with oauth authent.
I'll tell you the result of my reserch soon
Thanks
